### ISSUE TITLE
New rule `react-process-inline-styles`

### ISSUE THREAD

**Issue by loicraux** [Microsoft/tslint-microsoft-contrib#244]
What about a new `react-process-inline-styles` rule that would check that any usage of the `style={...}` attribute in React JSX (see [this page in React doc](https://facebook.github.io/react/tips/inline-styles.html)) uses a call to a processor function (which name would be given in the rule configuration) as a value ?

This would allow to make sure that React.CSSProperties object being used as inline styles in React components are processed (some autoprefixes are added, handling RTL, ...) before they are actually used by React when rendering.

```javascript
// given that the rule react-process-inline-styles with 'process' function specified in the
// rule configuration is enabled.

// KO
public render(): ReactTypes.ReactElement<ReactTypes.HTMLAttributes> {
    return <div style={someStyle}>foo bar</div>;
}

// OK
public render(): ReactTypes.ReactElement<ReactTypes.HTMLAttributes> {
    return <div style={process(someStyle)}>foo bar</div>;
}

// OK also :
public render(): ReactTypes.ReactElement<ReactTypes.HTMLAttributes> {
    return <div style={process(someStyle, { forceLtr: true })}>foo bar</div>;
}

// KO
public render(): ReactTypes.ReactElement<ReactTypes.HTMLAttributes> {
    return <div style={whatever(process(someStyle))}>foo bar</div>;
}
```

**Comment by HamletDRC** [comment]
Do we use some sort of processor today in MSE? I thought we used auto-prefixer for this sort of thing when dealing with raw CSS... this processor function would do the same thing as auto-prefixer except for inline styles... right? 

Sounds reasonable.

**Comment by loicraux** [comment]
We use postcss CSS postprocessor for our stylesheets (flipping things if necessary for RTL, prefixing some properties, and minifying) , but nothing for our inline styles, which causes many issues with RTL version of UI.
The goal would be to let developers write inline styles as they do today (thinking LTR), but force them to always "postprocess" them before using them in the  _style_ property.

**Comment by loicraux** [comment]
Closing this issue because it is too MS specific.
